PactNet/Mocks/MockHttpService/AdminHttpClient.cs

```diff
@@ -78,14 +77,18 @@ public async Task SendAdminHttpRequest(HttpVerb method, string path, IDictionary
             Dispose(request);
             Dispose(response);
 
-            if (path == "/pact" && headers != null && headers.ContainsKey("fileNameAndPath"))
+            if (responseStatusCode != HttpStatusCode.OK)
             {
-                File.WriteAllText(headers["fileNameAndPath"], responseContent);
+                throw new PactFailureException(responseContent);
             }
 
-            if (responseStatusCode != HttpStatusCode.OK)
+            if (!string.IsNullOrEmpty(responseContent))

```

Comment:
Nice to have:

IMHO it a little bit cleaner when you define local variable close to their usage. In your case you define response content at the top and use it almost at the bottom.  Additionally I think (I am not 100% sure) in our case ReadAsStringAsync will never return a null, so you can simplify code a bit. 

```
          var responseContent = string.Empty;

            if (response.Content != null)
            {
                responseContent = await response.Content.ReadAsStringAsync();
            }

            Dispose(request);
            Dispose(response);

            if (responseStatusCode != HttpStatusCode.OK)
            {
                throw new PactFailureException(responseContent);
            }

            return responseContent; //If for some reason you think it can be null you can use ?? - responseContent ?? string.Empty
```